enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/mtls/MtlsUtils.java

```diff
@@ -0,0 +1,119 @@
+import java.net.URLDecoder;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Map;
+
+/**
+ * Common functions for certificates.
+ */
+public class MtlsUtils {
+    public static boolean isCertificate(String cert) {
+        return cert.contains("BEGIN") && cert.contains("CERTIFICATE") && cert.contains("END");
+    }
+
+    public static String getCertContent(String cert) {
+        return getCertContent(cert, false);
+    }
+    public static String getCertContent(String cert, boolean isEncoded) {
+        if (isEncoded) {
+            try {
+                cert = URLDecoder.decode(cert, "UTF-8");
+            } catch (UnsupportedEncodingException e) {
+                throw new SecurityException(e);
+            }
+        }
+        String content = cert.replaceAll("-----BEGIN CERTIFICATE-----", "")
+                .replaceAll("-----END CERTIFICATE-----", "");
+        return content.trim();
+    }
+
+    public static KeyStore createTrustStore(Map<String, ByteString> clientCertMap) throws KeyStoreException {
+        KeyStore trustStore = KeyStore.getInstance("JKS");
+        try {
+            trustStore.load(null, null);
+        } catch (IOException | NoSuchAlgorithmException | CertificateException e) {
+            throw new SecurityException(e);
+        }
+        X509Certificate x509cert;
+        for (Map.Entry<String, ByteString> entry : clientCertMap.entrySet()) {
+            String alias = entry.getKey();
+            String cert = entry.getValue().toStringUtf8();
+            String certContent = getCertContent(cert);
+            try {
+                x509cert = getX509Cert(certContent);
+            } catch (java.security.cert.CertificateException e) {
+                throw new SecurityException(e);
+            }
+            trustStore.setCertificateEntry(alias, x509cert);
+        }
+        return trustStore;
+    }
+
+    public static X509Certificate getX509Cert(String certContent)
+            throws CertificateException {
+        CertificateFactory cf = CertificateFactory.getInstance("X.509");
+        byte[] bytes = Base64.decodeBase64(certContent);
+        X509Certificate x509Certificate;
+        try (ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(bytes)) {
+            x509Certificate =
+                    (X509Certificate) cf.generateCertificate(byteArrayInputStream);
+        } catch (IOException e) {
+            throw new CertificateException(e);
+        }
+        return x509Certificate;
+    }
+
+    public static boolean isCertificateExistsInTrustStore(X509Certificate certificate, KeyStore trustStore)
+            throws CertificateException {
+        String alias;
+        try {
+            alias = trustStore.getCertificateAlias(certificate);
```

Comment:
If the truststore is null, this could throw a null pointer exception?

Reply:
Solved